refactor - migrate simple components to stateless functional components [ready]

hannahstwinter opened the pull request:
Another PR migrating simple components to stateless functional components. There will probably be one more though it will only have two or so components and I'm stalling because the effects are trickling through to other spec files. 🙊

willcumbie commented on src/FormFieldValidationBlock.jsx:
```diff
@@ -1,15 +1,15 @@
+    <span className="rs-validation-block">
+      <i className="rs-validation-indicator"></i> { props.value }
+    </span>
+  ) : <noscript />;
```
Why noscript rather than null?

hannahstwinter replied:
Note section on https://facebook.github.io/react/docs/reusable-components.html#stateless-functions - "In React v0.14, stateless functional components were not permitted to return null or false (a workaround is to return a <noscript /> instead). This was fixed in React v15...". I think we're using 0.14.8?

hannahstwinter replied:
lol - it ate my noscript tag

willcumbie commented on test/FormFieldSpec.jsx:
```diff
@@ -1,72 +1,71 @@
 
-  it('adds the help message', () => {
-    let message;
+  // it('adds the help message', () => {
```
Uncomment these.

hannahstwinter replied:
wherps

willcumbie commented:
LGTM

j00k commented:
:ok_hand:

adampickeral commented:
:+1: 

for future PR's, commit messages should look a little more like this:
```
refactor(stateless components): migrate components to stateless components
// anything else needed to explain the change in more detail
```
https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit-message-format

hannahstwinter commented:
@adampickeral I rebased and fixed it. I meant to do that before now - thanks for reminding me! I think I forgot to do this on the earlier comparable PR.
